zooniverse/front-end-monorepo
# Add highlighter task

## Package
- lib-classifier

## Linked Issue and/or Talk Post
- towards #4326 
- based on [the PFE Highlighter task](https://github.com/zooniverse/Panoptes-Front-End/tree/master/app/classifier/tasks/highlighter)

## Describe your changes
- adds highlighter task component, model, annotation
- updates text viewer for highlights

## Open Issues
- [x] annotation `end` value incorrect - a51f1413b1ab9949424c8b6f6110818d0b56e9dd
- [x] selection delete button styling - 1a4ea6264b6ca216dc27e43feca083fe36619175

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - fe-project-branch - https://fe-project-branch.preview.zooniverse.org/projects/markbouslog/digileap-testing-production/classify/workflow/23429
  - local/app-project - https://local.zooniverse.org:3000/projects/markb-panoptes/digileap-testing-staging/classify/workflow/3538
  - local/lib-classifier - https://local.zooniverse.org:8080/?project=1952&workflow=3538
- What user actions should my reviewer step through to review this PR?
  1. highlight text in the text viewer
  2. click one of the highlighter task label buttons in the task area
  3. repeat steps 1-2 as necessary
  4. click Done button in task area 
- Which storybook stories should be reviewed?
  - http://localhost:6006/?path=/story/tasks-highlighter--default
  - http://localhost:6006/?path=/story/subject-viewers-singletextviewer--with-labeled-text
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Post-merge
